src/CircuitOptimizer.cpp

```diff
@@ -527,4 +670,107 @@ namespace qc {
+	{
+		//decompose SWAPS in three cnot and optionally in four H
+		auto it = qc.ops.begin();
+		while (it != qc.ops.end())
+		{
+			if ((*it)->isStandardOperation())
+			{
+				if ((*it)->getType() == qc::SWAP)
+				{
+					std::vector<unsigned short> targets = (*it)->getTargets();
+					it = qc.ops.erase(it);
+					it = qc.ops.insert(it, std::make_unique<StandardOperation>((*it)->getNqubits(), Control(targets[0]), targets[1], qc::X));
+					if (isDirectedArchitecture)
+					{
+						it = qc.ops.insert(it, std::make_unique<StandardOperation>((*it)->getNqubits(), targets[0], qc::H));
+						it = qc.ops.insert(it, std::make_unique<StandardOperation>((*it)->getNqubits(), targets[1], qc::H));
+						it = qc.ops.insert(it, std::make_unique<StandardOperation>((*it)->getNqubits(), Control(targets[0]), targets[1], qc::X));
+						it = qc.ops.insert(it, std::make_unique<StandardOperation>((*it)->getNqubits(), targets[0], qc::H));
+						it = qc.ops.insert(it, std::make_unique<StandardOperation>((*it)->getNqubits(), targets[1], qc::H));
+					}
+					else
+					{
+						it = qc.ops.insert(it, std::make_unique<StandardOperation>((*it)->getNqubits(), Control(targets[1]), targets[0], qc::X));
+					}
+					it = qc.ops.insert(it, std::make_unique<StandardOperation>((*it)->getNqubits(), Control(targets[0]), targets[1], qc::X));
+				}
+				else
+				{
+					++it;
+				}
+			}
+			else if ((*it)->isCompoundOperation())
+			{
+				auto compOp = dynamic_cast<qc::CompoundOperation *>((*it).get());
+				auto cit = compOp->begin();
+				while (cit != compOp->end())
+				{
+					auto cop = dynamic_cast<CompoundOperation *>(cit->get());
+					if (cop->isStandardOperation() && cop->getType() == qc::SWAP)
+					{
+						std::vector<unsigned short> targets = cop->getTargets();
+						cit = cop->erase(cit);
+						cit = cop->insert<StandardOperation>(cit, compOp->getNqubits(), Control(targets[0]), targets[1], qc::X);
+						if (isDirectedArchitecture)
+						{
+							cit = cop->insert<StandardOperation>(cit, compOp->getNqubits(), targets[0], qc::H);
+							cit = cop->insert<StandardOperation>(cit, compOp->getNqubits(), targets[1], qc::H);
+							cit = compOp->insert<StandardOperation>(cit, compOp->getNqubits(), Control(targets[0]), targets[1], qc::X);
+							cit = cop->insert<StandardOperation>(cit, compOp->getNqubits(), targets[0], qc::H);
+							cit = cop->insert<StandardOperation>(cit, compOp->getNqubits(), targets[1], qc::H);
+						}
+						else
+						{
+
+							cit = compOp->insert<StandardOperation>(cit, compOp->getNqubits(), Control(targets[0]), targets[1], qc::X);
+						}
+						cit = cop->insert<StandardOperation>(cit, compOp->getNqubits(), Control(targets[0]), targets[1], qc::X);
+					}
+					else
+					{
+						++cit;
+					}
+				}
+				++it;
+			}
+			else
+			{
+				++it;
+			}
+		}
+	}
+
+	void CircuitOptimizer::decomposeTeleport(QuantumComputation &qc){
+
+	}
+
 }
```

Comment:
Is there a reason for changing the formatting of all the brackets? This makes it quite hard to identify the actual changes in the code. Could you maybe revert this in order to reduce the amount of unnecessary whitespace change?